src/index.js

```diff
@@ -26,40 +26,73 @@ type Options = {
+
 const createServer = (options?: Options): Server => {
     const {port = 0, ssl, onResponseNotFound} = options || {};
     const mockingServer = createMockingServer({onResponseNotFound});
 
+    const sendResponse = (response, headers, code, content) => {
+        response.statusCode = code;
+        Object.keys(headers).forEach((h) => response.setHeader(h, headers[h]));
+        response.write(content);
+        response.end();
+    };
+
     const handleRequest = (request, response) => {
-        const form = new multiparty.Form();
-        form.parse(request, (err, formFields = {}, files) => {
-            const {method, url, headers} = request;
-            const {pathname, query = {}} = parseUrl(url, true);
-
-            if (pathname === '/__admin__/kill') {
-                response.write('killed');
-                response.end(() => {
-                    process.exit();
+        const {method, url, headers} = request;
+        const {pathname, query = {}} = parseUrl(url, true);
+        const contentType = request.headers['content-type'];
+
+        if (pathname === '/__admin__/kill') {
+            response.write('killed');
+            response.end(() => {
+                process.exit();
+            });
+        }
+
+        if (isMultipart(contentType)) {
+            const form = new multiparty.Form();
+            form.parse(request, (err, formFields = {}, files) => {
+                console.log(err, formFields, files);
+                const res = mockingServer.handle({
+                    method,
+                    urlPath: pathname || '',
+                    urlParams: query,
+                    headers,
+                    formFields,
                 });
-            }
-
-            const res = mockingServer.handle({
-                method,
-                urlPath: pathname || '',
-                urlParams: query,
-                headers,
-                formFields,
+
+                if (res) {
+                    const {headers: responseHeaders, content, statusCode} = res;
+
+                    sendResponse(response, responseHeaders, statusCode, content);
+                }
             });
+        } else {
+            let body = '';
 
-            if (res) {
-                const {headers: responseHeaders, content, statusCode} = res;
+            request.on('data', (data) => {
+                body += data;
+            });
 
-                response.statusCode = statusCode;
-                Object.keys(responseHeaders).forEach((h) => response.setHeader(h, responseHeaders[h]));
-                response.write(content);
-                response.end();
-            }
-        });
+            request.on('end', () => {
+                const res = mockingServer.handle({
+                    method,
+                    urlPath: pathname || '',
+                    urlParams: query,
+                    headers,
+                    formFields: body,
```

Comment:
shouldn't you parse the body here and transform the string into a ParamsBag ?

Reply:
I changed ParamsBags to be an object or a string (in case this is a plain text). I really don't know what is the best way to do this, as all this was broken for all post request, WDYT?